### PR TITLE
fix(blog): normalize inline authors socials

### LINKED ISSUE #10417: Blog author social platforms shorthands are not normalized for inline authors
### Description

When authors are provided inline (not in `authors.yml`) authors handle/username shorthand are not normalized to full URLs, leading to incorrect links.

### Reproducible demo

_No response_

### Steps to reproduce

Create a blog post

```yml
slug: first-blog-post
title: First Blog Post
authors: 
  - name: John Doe
    title: Software Engineer
    socials: 
      github: slorber
```

### Expected behavior

The author GitHub icon link should be `https://github.com/slorber` 

### Actual behavior

The author GitHub icon link shouldn't be not `http://localhost:3000/slorber`

### PR DESCRIPTION
**Pull request by OzakIOne**
## Motivation

Fix #10417

## Test Plan

Local & CI

### Test links

Deploy preview: https://deploy-preview-10424--docusaurus-2.netlify.app/

### CODE REVIEW

File: packages/docusaurus-plugin-content-blog/src/authors.ts [hunk 1]
```diff
@@ -117,12 +119,16 @@ ${Object.keys(authorsMap)
       ...getAuthorsMapAuthor(frontMatterAuthor.key),
       ...frontMatterAuthor,
     };
+    const normalizedSocials = normalizeSocials(frontMatterAuthor.socials ?? {});
```
Comment (slorber) [inline packages/docusaurus-plugin-content-blog/src/authors.ts:122]: This works but doesn't look like the best place to add this logic.

I'd add it there instead:

CleanShot 2024-08-20 at 12 15 02@2x

**Review by slorber — CHANGES_REQUESTED** [review]
Need to move implementation to `normalizeAuthor` + add a few new tests for `getBlogPostAuthors`

File: packages/docusaurus-plugin-content-blog/src/authors.ts [hunk 2]
```diff
@@ -85,8 +85,14 @@ function getFrontMatterAuthors(params: AuthorsParam): Author[] {
...
       }
-
-      return authorInput;
+      const normalizedSocials = normalizeSocials(authorInput.socials ?? {});
+
+      return {
+        ...authorInput,
+        ...(Object.keys(normalizedSocials).length > 0 && {
+          socials: normalizedSocials,
+        }),
```
Comment (slorber) [inline packages/docusaurus-plugin-content-blog/src/authors.ts:94]: if socials = {} I wonder if we shouldn't keep socials = {} in the output

File: packages/docusaurus-plugin-content-blog/src/authors.ts [hunk 3]
```diff
@@ -84,7 +85,10 @@ function getFrontMatterAuthors(params: AuthorsParam): Author[] {
         // becoming a name and may end up unnoticed
         return {key: authorInput};
       }
-      return authorInput;
+      return {
+        ...authorInput,
+        socials: normalizeSocials(authorInput.socials ?? {}),
```
Comment (slorber) [inline packages/docusaurus-plugin-content-blog/src/authors.ts:90]: Here if author front matter has no social key, the result will be `social: {}`.

If the key was absent before normalization, we'd rather keep it absent after normalization (unless it's an explicit goal of the normalization to always have an object for empty (in which case the normalized type shouldn't allow null/undefined anymore)

File: packages/docusaurus-plugin-content-blog/src/__tests__/authors.test.ts [hunk 4]
```diff
@@ -268,11 +268,132 @@ describe('getBlogPostAuthors', () => {
...
-  it('can read authors Author', () => {
+  it('can read empty socials', () => {
     expect(
       getBlogPostAuthors({
         frontMatter: {
-          authors: {name: 'Sébastien Lorber', title: 'maintainer'},
+          authors: {
+            name: 'Sébastien Lorber',
+            title: 'maintainer',
+            socials: {},
```
Comment (slorber) [inline packages/docusaurus-plugin-content-blog/src/__tests__/authors.test.ts:278]: We also need tests for undefined/null/absent social key

As you will see your implementation might not do what you expect

File: project-words.txt [hunk 5]
```diff
@@ -330,8 +330,11 @@ showinfo
 Sida
 Simen
 slorber
+Slorber
 sluggifies
 sluggify
+socialnull
+socialnumber
```
Comment (slorber) [inline project-words.txt:337]: Try to get rid of these by using camel case `socialNull` for example, both words exist independently

**Review by slorber — CHANGES_REQUESTED** [review]

File: packages/docusaurus-plugin-content-blog/src/authors.ts (hunk 3 above)
Comment (slorber) [reply packages/docusaurus-plugin-content-blog/src/authors.ts:90]: I don't think we plan render those differently.

I also like to normalize to `{}` for convenience of frontend code, although technically this slightly increases bundle size / memory by creating useless empty objects when we could use optional chaining. Maybe always normalizing to undefined would be better? That's really a detail so I'm fine with any of those.

Comment (Josh-Cena) [reply packages/docusaurus-plugin-content-blog/src/authors.ts:90]: I don't think neither of those would be a concern when there aren't thousands of authors (by which time the blog posts' metadata is a far bigger concern).

**Review by slorber — APPROVED** [review]
LGTM thanks 👍

**Comment by slorber** [comment]
oups, updated the branch and now we have test failures 😄
